x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/OptimizerTests.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.optimizer;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.eql.analysis.Analyzer;
+import org.elasticsearch.xpack.eql.analysis.PreAnalyzer;
+import org.elasticsearch.xpack.eql.analysis.Verifier;
+import org.elasticsearch.xpack.eql.expression.function.EqlFunctionRegistry;
+import org.elasticsearch.xpack.eql.parser.EqlParser;
+import org.elasticsearch.xpack.ql.expression.FieldAttribute;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.And;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.Not;
+import org.elasticsearch.xpack.ql.expression.predicate.nulls.IsNotNull;
+import org.elasticsearch.xpack.ql.expression.predicate.nulls.IsNull;
+import org.elasticsearch.xpack.ql.expression.predicate.regex.Like;
+import org.elasticsearch.xpack.ql.index.EsIndex;
+import org.elasticsearch.xpack.ql.index.IndexResolution;
+import org.elasticsearch.xpack.ql.plan.logical.Filter;
+import org.elasticsearch.xpack.ql.plan.logical.LogicalPlan;
+import org.elasticsearch.xpack.ql.type.EsField;
+import org.elasticsearch.xpack.ql.type.TypesTests;
+
+import java.util.Map;
+
+public class OptimizerTests extends ESTestCase {
+
+
+    private static final String INDEX_NAME = "test";
+    private EqlParser parser = new EqlParser();
+    private IndexResolution index = loadIndexResolution("mapping-default.json");
+
+    private static Map<String, EsField> loadEqlMapping(String name) {
+        return TypesTests.loadMapping(name);
+    }
+
+    private IndexResolution loadIndexResolution(String name) {
+        return IndexResolution.valid(new EsIndex(INDEX_NAME, loadEqlMapping(name)));
+    }
+
+    private LogicalPlan accept(IndexResolution resolution, String eql) {
+        PreAnalyzer preAnalyzer = new PreAnalyzer();
+        Analyzer analyzer = new Analyzer(new EqlFunctionRegistry(), new Verifier());
+        Optimizer optimizer = new Optimizer();
+        return optimizer.optimize(analyzer.analyze(preAnalyzer.preAnalyze(parser.createStatement(eql), resolution)));
+    }
+
+    private LogicalPlan accept(String eql) {
+        return accept(index, eql);
+    }
+    
+    public void testIsNull() {
+        for (String q : new String[]{"foo where command_line == null", "foo where null == command_line"}) {
```

Comment:
It's uncommon to inline a constant declaration within a for loop mainly because an increase in the number of arguments makes it impractical.

Typically, declaring a method with varags and then passing the arguments inside its invocation or declaring the list/array in a variable is the way to go. (`Arrays.asList` is used a lot, potentially with a static imports - there's also `List.of` but that is not available in Java 8, used by ES 7.x).